test/e2e/ComplexShrink.spec.ts

```diff
@@ -0,0 +1,101 @@
+    // . . . √ [1610612157,2147483641]
+    // . . . √ [1879047517,2147483641]
+    // . . . √ ...
+    // . . . √ [2147482621,2147483641]
+    // . . . × [2147482749,2147483641]
+    // ...                              <--- and so on and so forth until we reach 1000,1010
+    //
+    // While it was able to reach and report the precise minimal failing case,
+    // the shrinker was too slow to be useful (given the range of values generated).
+    const MaximalValue = 1000000;
+    const MaximalNumShrinksAllowed = 1000;
+    const out = fc.check(
+      fc.property(fc.nat(MaximalValue), fc.nat(MaximalValue), (a: number, b: number) => {
+        if (a < 1000) return true;
+        if (b < 1000) return true;
+        if (b < a) return true;
+        if (Math.abs(a - b) < 10) return true;
+        return b - a >= 1000;
+      }),
+      // We increase numRuns to avoid flakiness in the CI
+      { seed, numRuns: 1000 }
+    );
+
+    // > should find the failure
+    expect(out.failed).toBe(true);
+
+    // > should find a barely minimal failing case
+    // Such failing case is a local minimum. Reaching a difference
+    // of 10 already proves that the shrinker did part of the job.
+    const minimal = out.counterexample!;
+    expect(minimal[1] - minimal[0]).toBe(10);
+
+    // > should find the minimal failing case
+    // Reaching this minimal case requires advanced shrinker logic
+    // It implies a shrinker able to shrink on two values at the same time
+    expect(minimal).toEqual([1000, 1010]);
+    expect(out.numShrinks).toBeLessThanOrEqual(MaximalNumShrinksAllowed);
+  });
+  it('Should shrink two integers linked by a symmetric relation', () => {
+    // Very similar to the case above except that this time
+    // if (a, b) fails, (b, a) will also fail.
+    const MaximalValue = 1000000;
+    const MaximalNumShrinksAllowed = 1000;
+    const out = fc.check(
+      fc.property(fc.nat(MaximalValue), fc.nat(MaximalValue), (a: number, b: number) => {
+        if (a < 1000) return true;
+        if (b < 1000) return true;
+        if (Math.abs(a - b) < 10) return true;
+        return Math.abs(a - b) >= 1000;
+      }),
+      // We increase numRuns to avoid flakiness in the CI
+      { seed, numRuns: 1000 }
+    );
+
+    // > should find the failure
+    expect(out.failed).toBe(true);
+
+    // > should find a barely minimal failing case
+    // Such failing case is a local minimum. Reaching a difference
+    // of 10 already proves that the shrinker did part of the job.
+    const [a, b] = out.counterexample!;
+    const minimal = a < b ? [a, b] : [b, a];
+    expect(Math.abs(minimal[1] - minimal[0])).toBe(10);
+
+    // > should find the minimal failing case
+    // Reaching this minimal case requires advanced shrinker logic
+    // It implies a shrinker able to shrink on two values at the same time
+    expect(minimal).toEqual([1000, 1010]);
+    expect(out.numShrinks).toBeLessThanOrEqual(MaximalNumShrinksAllowed);
```

Comment:
```suggestion
    const MaximalValue = 1000000;
    const out = fc.check(
      fc.property(fc.nat(MaximalValue), fc.nat(MaximalValue), (a: number, b: number) => {
        if (a < 1000) return true;
        if (b < 1000) return true;
        if (Math.abs(a - b) < 10) return true;
        return Math.abs(a - b) >= 1000;
      }),
      // We increase numRuns to avoid flakiness in the CI
      { seed, numRuns: 1000 }
    );

    // > should find the failure
    expect(out.failed).toBe(true);

    // > should find a barely minimal failing case
    // Such failing case is a local minimum. Reaching a difference
    // of 10 already proves that the shrinker did part of the job.
    const [a, b] = out.counterexample!;
    const minimal = a < b ? [a, b] : [b, a];
    expect(Math.abs(minimal[1] - minimal[0])).toBe(10);

    // > should find the minimal failing case
    // Reaching this minimal case requires advanced shrinker logic
    // It implies a shrinker able to shrink on two values at the same time
    expect(minimal).toEqual([1000, 1010]);
```